File: ml/src/main/scala/ai/h2o/sparkling/ml/params/HasBaseAlgorithms.scala
package ai.h2o.sparkling.ml.params

import ai.h2o.sparkling.H2OFrame
import ai.h2o.sparkling.ml.algos.H2OAlgorithm
import hex.Model

trait HasBaseAlgorithms extends H2OAlgoParamsBase {

  private val base_algorithms = new NullableAlgoArrayParam(this, "baseAlgorithms", "An array of base algorithms")

  setDefault(base_algorithms -> null)

  def getBaseAlgorithms(): Array[H2OAlgorithm[_ <: Model.Parameters]] = $(base_algorithms)

  def setBaseAlgorithms(value: Array[H2OAlgorithm[_ <: Model.Parameters]]): this.type = set(base_algorithms, value)

  private[sparkling] def getBaseAlgorithmsParam(trainingFrame: H2OFrame): Map[String, Any] = {
    // the base_algorithms parameter isn't used by H2O backend
    Map.empty
  }
}

File: py-scoring/src/ai/h2o/sparkling/ml/params/HasBaseAlgorithms.py
from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters
from pyspark.ml.param import *


class HasBaseAlgorithms(Params):
    baseAlgorithms = Param(
        Params._dummy(),
        "baseAlgorithms",
        "Algorithms used by a meta learner.",
        H2OTypeConverters.toNullableListJavaObject())

    def getBaseAlgorithms(self):
        return self.getOrDefault(self.baseAlgorithms)

    def setBaseAlgorithms(self, algos):
        return self._set(baseAlgorithms=algos)
